tests/integration/test_connection.py

```diff
@@ -0,0 +1,77 @@
+import pytest
+from tests.integration.utils import *
+
+
+def test_anonymous_connection_endpoints():
+    anonymous_connect = get_anonymous_connection()
+    repo, auth, file = get_test_endpoints()
+    assert repo == anonymous_connect.repo_endpoint
+    assert auth == anonymous_connect.auth_endpoint
+    assert file == anonymous_connect.file_endpoint
+
+
+def test_user_connection_endpoints():
+    synapse_connection = get_test_user_connection()
+    repo, auth, file = get_test_endpoints()
+    assert repo == synapse_connection.repo_endpoint
+    assert auth == synapse_connection.auth_endpoint
+    assert file == synapse_connection.file_endpoint
+
+
+def test_anonymous_connection_get():
+    anonymous_connect = get_anonymous_connection()
+    version = anonymous_connect.get("/version")
+    assert version['version'] is not None
+
+
+def test_anonymous_connection_post():
+    anonymous_connect = get_anonymous_connection()
+    _, auth, _ = get_test_endpoints()
+    username, password = get_test_credentials()
+    login_request = {'username': username,
+                     'password': password}
+    response = anonymous_connect.post("/login", endpoint=auth, request_body=login_request)
+    assert response['sessionToken'] is not None
+
+
+def test_anonymous_connection_put():
+    anonymous_connect = get_anonymous_connection()
+    _, auth, _ = get_test_endpoints()
+    with pytest.raises(SynapseUnauthorizedError):
```

Comment:
add comment mentioning that synapse does not have any put methods accessible by anonymous users